File: android/app/src/main/java/com/alandaitch/anden/data/model/EcobiciStation.kt
package com.alandaitch.anden.data.model

import com.alandaitch.anden.util.GeoPoint
import java.time.Instant

// Estación de EcoBici (GBFS). Une stationInformation + stationStatus.
data class EcobiciStation(
    val id: String,
    val name: String,
    val lat: Double,
    val lng: Double,
    val capacity: Int,
    val bikesMechanical: Int,
    val bikesEbike: Int,
    val bikesTotal: Int,
    val docksAvailable: Int,
    val status: String,
    val lastReported: Instant
) {
    val coordinate: GeoPoint get() = GeoPoint(lat, lng)

    // Nombre sin el prefijo numérico. "002 - Retiro I" -> "Retiro I".
    val displayName: String
        get() {
            val parts = name.split("-", limit = 2)
            if (parts.size == 2 && parts[0].trim().toIntOrNull() != null) {
                return parts[1].trim()
            }
            return name.trim()
        }
}

File: Shared/Models/EcobiciStation.swift
import Foundation
import CoreLocation

// Estación de EcoBici (GBFS). Une stationInformation + stationStatus.
struct EcobiciStation: Identifiable, Hashable {
    let id: String
    let name: String
    let lat: Double
    let lng: Double
    let capacity: Int
    let bikesMechanical: Int
    let bikesEbike: Int
    let bikesTotal: Int
    let docksAvailable: Int
    let status: String
    let lastReported: Date

    var coordinate: CLLocationCoordinate2D {
        CLLocationCoordinate2D(latitude: lat, longitude: lng)
    }

    // Nombre sin el prefijo numérico. "002 - Retiro I" -> "Retiro I".
    var displayName: String {
        let parts = name.split(separator: "-", maxSplits: 1, omittingEmptySubsequences: false)
        if parts.count == 2,
           Int(parts[0].trimmingCharacters(in: .whitespaces)) != nil {
            return parts[1].trimmingCharacters(in: .whitespaces)
        }
        return name.trimmingCharacters(in: .whitespaces)
    }
}
